_posts/2021-08-26-geometric-deep-learning/geometric_deep_learning.Rmd

```diff
@@ -0,0 +1,125 @@
+
+### Graphs and GNNs
+
+Another type of domain, underlying many scientific and non-scientific applications, are graphs. Here, we are going to be a lot more brief. One reason is that so far, we have not had many posts on deep learning on graphs, so to the readers of this blog, the topic may seem fairly abstract. The other reason is complementary: That state of affairs is exactly something we'd like to see changing. Once we write more about graph DL, occasions to talk about respective concepts will be plenty.
+
+In a nutshell, though, the dominant type of invariance in graph DL is permutation equivariance. Permutation, because when you stack a node and its features in a matrix, it doesn't matter whether node one is in row three or row fifteen. Equivariance, because once you *do* permute the nodes, you also have to permute the adjacency matrix, the matrix that captures which node is linked to what other nodes. This is very different from what holds for images: We can't just randomly permute the pixels.
+
+### Sequences and RNNs
+
+With RNNs, we are going be very brief as well, although for a different reason. My impression is that so far, this area of research -- meaning, GDL as it relates to sequences -- has not received too much attention yet, and (maybe) for that reason, seems of lesser impact on real-world applications.
+
+In a nutshell, the authors refer two types of symmetry: First, translation-invariance, as long as a sequence is left-padded for a sufficient number of steps. (This is due to the hidden units having to be initialized *somehow*.) This holds for RNNs in general.
+
+Second, *time warping*: If a network can be trained that correctly works on a sequence measured on some time scale, there is another network, of the same architecture but likely with different weights, that will work equivalently on re-scaled time. This invariance only applies to *gated* RNNs, such as the LSTM.
+
+### What's next?
+
+At this point, we conclude this conceptual introduction. If you want to learn more, and are not too scared by the math, definitely check out the book. (I'd also say it lends itself well to incremental understanding, as in, iteratively going back to some details once one has acquired more background.)
+
+The other desiderate certainly is practice. There is an intimate connection between GDL and deep learning on graphs; which is one reason we're hoping to be able to feature the latter more frequently in the future. The other is the wealth of interesting applications that take graphs as their input. Until then, thanks for reading!
```

Comment:
"desiderate" is a pretty archaic English word. Maybe consider rephrasing the first sentence in this paragraph?